jvm/src/main/scala/com/avast/cloud/metrics/datadog/jvm/JvmMonitoring.scala

```diff
@@ -0,0 +1,54 @@
+package com.avast.cloud.metrics.datadog.jvm
+
+import java.time.Duration
+import java.util.concurrent.{ Executors, ScheduledExecutorService, ThreadFactory, TimeUnit }
+
+import cats.effect.{ Effect, IO, Resource, Sync }
+import com.avast.cloud.metrics.datadog.api.MetricFactory
+
+object JvmMonitoring {
+  type ErrorHandler[F[_]] = Throwable => F[Unit]
+
+  case class Config(initialDelay: Duration = Duration.ofMillis(0),
+                    delay: Duration = Duration.ofMinutes(1),
+                    schedulerThreadName: String = "datadog-jvm-reporter")
+
+  def default[F[_]: Effect](factory: MetricFactory[F]): Resource[F, Unit] =
+    configured(factory, Config(), defaultErrorHandler)
+
+  def configured[F[_]: Effect](factory: MetricFactory[F],
+                               config: Config,
+                               errorHandler: ErrorHandler[F]): Resource[F, Unit] = {
+    val F = Effect[F]
+    Resource.make(F.delay(makeScheduler(config)))(s => F.delay(s.shutdown())).evalMap { scheduler =>
+      F.delay {
+        val reporter = new JvmReporter[F](factory)
+        scheduler.scheduleWithFixedDelay(runnable(reporter, errorHandler andThen F.toIO),
+                                         config.initialDelay.toNanos,
+                                         config.delay.toNanos,
+                                         TimeUnit.NANOSECONDS)
+      }
+    }
+  }
+
+  private def makeScheduler(config: Config): ScheduledExecutorService =
+    Executors.newScheduledThreadPool(1, (r: Runnable) => new Thread(r, config.schedulerThreadName))
+
+  private def runnable[F[_]: Effect](reporter: JvmReporter[F], errorHandler: ErrorHandler[IO]): Runnable =
+    () => {
+      val syncIO = Effect[F].runAsync(reporter.collect) { res =>
```

Comment:
Why not `Effect[F].runAsync(reporter.collect).handleError(errorHandler).unsafeRunSync()` (requires `import cats.syntax.applicativeError._`

Reply:
didn't know about it

Reply:
are you sure this works?  runAsync requires two arguments

Reply:
nvm i figured it out